Improve README and remove the remaining references to the laboratory-operator (#401)
netgroup-polito/CrownLabs: README.md

# CrownLabs

CrownLabs is a set of services designed to deliver **remote computing labs** through **per-user environments**, based either on virtual machines or lightweight containers.

> Instructors can provision a set of environments (e.g. virtual machines), properly equipped with the software required for a given lab (e.g., compilers, simulation software, etc).

Reviewer: ```suggestion
Instructors can provision a set of environments (e.g., VMs), properly installed with the software required for a given lab (e.g., compilers, simulation software, etc).
```

## Architecture

CrownLabs relies on two major high-level components:
* **The Backend Business Logic**, which provides the different CrownLabs functionalities and is implemented by custom Kubernetes operators (e.g. the Instance Operator);
* **The Frontend Dashboard**, which interacts with the Kubernetes API Server and exposes the different CrownLabs custom resources through a graphical interface.

A high-level representation of the main architectural building blocks composing CrownLabs is given by the following figure.

> Please notice that, for the sake of clarity, the figure depicts the elements essential for the provision of the actual service (i.e. remote computing labs), while leaving out with those more low-level or associated with the cluster operation (e.g monitoring).

Reviewer: ```suggestion
Please notice that, for the sake of clarity, the figure depicts only the most important elements for the provision of the actual service (i.e., remote computing labs), while omitting low-level components and the ones associated with the cluster operation (e.g monitoring).
```

![CrownLabs High-Level Architecture](documentation/architecture.svg)

## Backend Business Logic

The backend business logic providing the different CrownLabs functionalities is implemented by custom Kubernetes operators, while the data model is defined by means of CRDs.
Specifically, the main backend components are:

> * the Instance Operator, which implements the logic to spawn new environments starting from predefined templates;

Reviewer: ```suggestion
* the **Instance Operator**, which implements the logic to spawn new environments starting from predefined templates;
```

> * the Tenant Operator, which automates the management of CrownLabs users (i.e. tenants) and groups (i.e. workpaces);

Reviewer: ```suggestion
* the **Tenant Operator**, which automates the management of CrownLabs users (i.e. tenants) and groups (i.e. workspaces);
```

> * the Bastion Operator, which configures an SSH bastion to provide command-line access to the environments.

Reviewer: ```suggestion
* the **Bastion Operator**, which configures an SSH bastion to provide command-line access to the environments instead of the web-based GUI.
```

Furthermore, some additional components are leveraged to simplify and automate companion tasks, such as listing the available images and deleting stale environments.

For more information regarding the CrownLabs backend, as well as for the deployment and configuration instructions, please refer to the corresponding [README](./operators/README.md).

## Frontend Dashboard

The frontend dashboard is the component responsible for providing access to the CrownLabs custom resources through an easy to use graphical interface.
It allows final users to spawn new environments, as well as to define new templates.
Additionally, privileged users can create, update and delete both tenant and workspace resources, effectively managing the permissions granted to access the system.

The CrownLabs dashboard builds on top of [LiqoDash](https://github.com/liqotech/dashboard), a dynamic and general purpose dashboard designed to display and interact with Kubernetes resources.

> Specifically, it can be thought as a graphical kubectl, which directly interacts with the Kubernetes API server and provides access to both native and custom resources (i.e. those defined through CRDs).

Reviewer: ```suggestion
Specifically, it can be thought as a graphical `kubectl`, which directly interacts with the Kubernetes API server and provides access to both native and custom resources (i.e. those defined through CRDs).
```
